game-core/src/main/java/games/strategy/engine/lobby/moderator/toolbox/tabs/ModeratorsTabModel.java

```diff
@@ -1,22 +1,69 @@
 package games.strategy.engine.lobby.moderator.toolbox.tabs;
 
+import java.time.Instant;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
+import org.triplea.http.client.moderator.toolbox.ModeratorToolboxClient;
 
-/**
- * TODO: WORK IN PROGRESS!.
- */
+import games.strategy.triplea.settings.ClientSetting;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+
+@AllArgsConstructor
 class ModeratorsTabModel {
 
-  /**
-   * Indicates current user can add or remove moderators.
-   */
-  boolean isModeratorAdmin() {
-    return true;
+  private final ModeratorToolboxClient moderatorToolboxClient;
+  @Getter
+  private final boolean isSuperMod;
+
+  ModeratorsTabModel(final ModeratorToolboxClient moderatorToolboxClient) {
+    this.moderatorToolboxClient = moderatorToolboxClient;
+    isSuperMod = moderatorToolboxClient.isCurrentUserSuperMod(ClientSetting.moderatorApiKey.getValueOrThrow());
+  }
+
+  // TODO: test-me
+  List<List<String>> getModeratorList() {
+    return moderatorToolboxClient
+        .fetchModeratorList(ClientSetting.moderatorApiKey.getValueOrThrow())
+        .stream()
+        .map(modInfo -> isSuperMod
+            ? Arrays.asList(
+                modInfo.getName(),
+                Optional.ofNullable(modInfo.getLastLogin())
+                    .map(Instant::toString)
+                    .orElse(""),
+                "Generate API-Key",
+                "Remove Mod",
+                "Add Super-Mod")
+            : Arrays.asList(
+                modInfo.getName(),
+            Optional.ofNullable(modInfo.getLastLogin())
+                  .map(Instant::toString)
+                .orElse("")))
```

Comment:
Like mentioned earlier in the data structure comment: Having different types here would make this more readable.

Reply:
It would be cool to have a `JTable<ObjectType>`  where it is initialized to have a mapper that could convert an `ObjectType` to a table row. I think that is a nice-to-have, probably better to ensure that the needed controls are present and that there are no errors in the conversion, nor is there any missing data. I do like that idea/thinking. I wonder if it's worth it long-term and jives with a javafx API. My hope is that we can convert to swing builders and then change the underlying implementation of the builds to produce JavaFx components instead. That way we are not updating any code outside of swing-lib when migrating away from swing.